src/feat/online-feature.cc

```diff
@@ -41,20 +41,37 @@ template<class C>
 void OnlineGenericBaseFeature<C>::AcceptWaveform(BaseFloat sampling_rate,
                                                  const VectorBase<BaseFloat> &waveform) {
   BaseFloat expected_sampling_rate = computer_.GetFrameOptions().samp_freq;
-  if (sampling_rate != expected_sampling_rate)
-    KALDI_ERR << "Sampling frequency mismatch, expected "
-              << expected_sampling_rate << ", got " << sampling_rate;
-  if (waveform.Dim() == 0)
+  bool allow_downsample = computer_.GetFrameOptions().allow_downsample;
```

Comment:
This won't give the right results if you call this multiple times with separate pieces of waveform.
You'll have to add a class member of type LinearResample*.  That object retains all the state necessary to do this right when different pieces of waveform are being appended together.  When the user says the waveform is done, you can, I think, call its Resample() function with an empty piece of waveform and flush == true.

You should probably add a function InputSampleRate() to the LinearResample object so that you can check the user doesn't call AcceptWaveform() with different sampling_rate values on different calls.  It's this class's responsibility to make sure that the sampling_rate values provided on different calls are all the same, so make sure if that is done, an error is raised and it doesn't just silently do the wrong thing in that case.

Reply:
@danpovey, I wonder if this commit contains what you asked me!
I did simple test in online setup and it works properly, but probably someone should test it.